doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -344,85 +343,96 @@ Rewriting with definitional equality
-conclusion.
-
-For backward compatibility, the notation :n:`in {+ @ident}` performs
-the conversion in hypotheses :n:`{+ @ident}`.
-
-.. tacn:: cbv {? @strategy_flag }
-          lazy {? @strategy_flag }
-   :name: cbv; lazy
-
-   These parameterized reduction tactics apply to any goal and perform
-   the normalization of the goal according to the specified flags. In
-   correspondence with the kinds of reduction considered in Coq namely
-   :math:`\beta` (reduction of functional application), :math:`\delta`
-   (unfolding of transparent constants, see :ref:`vernac-controlling-the-reduction-strategies`),
-   :math:`\iota` (reduction of
-   pattern matching over a constructed term, and unfolding of :g:`fix` and
-   :g:`cofix` expressions) and :math:`\zeta` (contraction of local definitions), the
-   flags are either ``beta``, ``delta``, ``match``, ``fix``, ``cofix``,
-   ``iota`` or ``zeta``. The ``iota`` flag is a shorthand for ``match``, ``fix``
-   and ``cofix``. The ``delta`` flag itself can be refined into
-   :n:`delta [ {+ @qualid} ]` or :n:`delta - [ {+ @qualid} ]`, restricting in the first
-   case the constants to unfold to the constants listed, and restricting in the
-   second case the constant to unfold to all but the ones explicitly mentioned.
-   Notice that the ``delta`` flag does not apply to variables bound by a let-in
-   construction inside the :n:`@term` itself (use here the ``zeta`` flag). In
-   any cases, opaque constants are not unfolded (see :ref:`vernac-controlling-the-reduction-strategies`).
-
-   Normalization according to the flags is done by first evaluating the
+   I didn't see how comparing these to "change" was very helpful, so I
+   dropped that sentence, too.
+
+.. tacn:: lazy {? @strategy_flag } {? @occurrences }
+          cbv {? @strategy_flag } {? @occurrences }
+
+   .. insertprodn red_expr pattern_occs
+
+   .. prodn::
+      red_expr ::= lazy {? @strategy_flag }
+      | cbv {? @strategy_flag }
+      | compute {? @delta_flag }
+      | vm_compute {? {| @reference_occs | @pattern_occs } }
+      | native_compute {? {| @reference_occs | @pattern_occs } }
+      | red
+      | hnf
+      | simpl {? @delta_flag } {? {| @reference_occs | @pattern_occs } }
+      | cbn {? @strategy_flag }
+      | unfold {+, @reference_occs }
+      | fold {+ @one_term }
+      | pattern {+, @pattern_occs }
+      | @ident
+      delta_flag ::= {? - } [ {+ @reference } ]
+      strategy_flag ::= {+ @red_flag }
+      | @delta_flag
+      red_flag ::= beta
+      | iota
+      | match
+      | fix
+      | cofix
+      | zeta
+      | delta {? @delta_flag }
+      reference_occs ::= @reference {? at @occs_nums }
+      pattern_occs ::= @one_term {? at @occs_nums }
+
+   Normalize the goal as specified by the :n:`@strategy_flag`\s. The flags
+   include:
+
+   `beta`
+     :term:`beta-reduction` of functional application
+
+   :n:`@delta_flag` and :n:`delta {? @delta_flag }`
```

Comment:
I would only put `delta` here. Then, in the indented description, I would make a special case for `delta_flag` provided alone.

Reply:
Take a look at what I did.  BTW, in what order should the `@reductions` be listed here and in the conversions chapter?  If I go by Greek alphabetical order, it should be beta, delta, zeta, iota, though I expect English readers would expect to see them in alpha order using the Romanized names.

Reply:
I don't have a strong opinion on the ordering. The Roman ordering makes sense to me, unless we want to privilege another ordering to make the presentation better.

Reply:
However, the ordering of the explanation should be the same as in the grammar (`reduction` nonterminal).